lib/parser/source/tree_rewriter.rb

```diff
@@ -326,6 +318,24 @@ def transaction
         @in_transaction = previous
       end
 
+      ##
+      # Provides access to a diagnostic engine.
+      # By default outputs diagnostic to $stderr
+      #
+      def self.default_diagnostics
+        @default_diagnostics ||= Diagnostic::Engine.new.tap do |engine|
+          engine.consumer = -> diag { $stderr.puts diag.render }
+        end
+      end
+
+      ##
+      # Provides access to a diagnostic engine.
+      # By default: self.class.default_diagnostics
+      #
+      def diagnostics
```

Comment:
I guess it still could be set in the constructor. Having initialization in a single place (like it was before) seems to be more readable.

Reply:
You're right, it would be in the constructor if it didn't need to be dup'ed, but it does...

Reply:
Yes, but even if you need `.dup` you still can do it in the constructor. Initialization of this field doesn't have to be lazy